File: 13-divide-by-zero-with-loop/main.cpp
/*
  Ernest Izdebski, 2022-09-14

  A program that accepts two numbers from the keyboard and divides the first by
  the second, ensuring the second is not zero with a while loop.
*/

#include <iostream>
using namespace std;

int main() {
  // define variables
  double divisor;
  double dividend;
  
  // populate variables
  cout << "Input divisor:  ";
  cin >> divisor;
  cout << "Input dividend: ";
  cin >> dividend;
  
  while (dividend == 0) {
    cout << "error: cannot divide by zero" << endl;
    cout << "Input dividend: ";
    cin >> dividend;
  }

  cout << divisor << " / " << dividend << " = " << (divisor / dividend) << endl;
}

File: 13-divide-by-zero-with-loop/src/main.rs
mod ernie;

fn main() {
    // define variables
    let x: f32 = ernie::input("input 1st number: ");
    let mut y: f32 = ernie::input("input 2nd number: ");

    // assert y != 0
    while y == 0.0 {
        println!("error: divisor cannot be zero");
        y = ernie::input("re-input 2nd number: ");
    }

    // display division
    println!("{} / {} = {}", x, y, x / y);
}
